generator/internal/sidekick/command.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sidekick
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+	"text/template"
+)
+
+// A Command is an implementation of a sidekick command
+// like sidekick generate
+type Command struct {
+	action           func(rootConfig *Config, cmdLine *CommandLine) error
```

Comment:
nit: I normally see this in Go named as `run`

Reply:
The command struct already has a `run` function that calls this after some validation.

Reply:
I don't think I see the need for the `run` function, so I would recommend deleting that and renaming this to run. If you think it is necessary though feel free to leave it, and we can always change things in a follow up.